src/python/WMCore/MicroService/MSPileup/DataStructs/MSPileupObj.py

```diff
@@ -107,11 +109,13 @@ def validate(self, pdict=None):
                 msg = f"Failed to validate: {key}, expect data-type {stype} got type {dtype}"
                 self.logger.error(msg)
                 return False, msg
-            if key == 'pileupName':
+            if key == 'pileupName' or (key == 'customName' and val != ''):
                 try:
                     dataset(val)
                 except AssertionError:
                     msg = f"pileupName value {val} does not match dataset pattern"
+                    if key == 'customName':
```

Comment:
These 2 lines could be removed with a parametrized message, e.g.: `                    msg = f"{key} value {val} does not match dataset pattern"`